common/persistence/history_manager.go

```diff
@@ -532,10 +532,17 @@ func (m *executionManagerImpl) NewHistoryBranch(
 	request *NewHistoryBranchRequest,
 ) (*NewHistoryBranchResponse, error) {
 
-	resp := &NewHistoryBranchResponse{}
-	var err error
-	resp.BranchToken, err = NewHistoryBranchToken(request.TreeId)
-	return resp, err
+	req := &NewHistoryBranchRequest{
```

Comment:
nit: is this the same as the request parameter? resp is also the same?

Reply:
good catch! For the longest time I actually had a parallel `Internal-` prefixed struct and decided last minute that was unnecessary for this particular function. I'll clean this up